pkg/e2/client.go

```diff
@@ -129,11 +131,16 @@ func (c *e2Client) Subscribe(ctx context.Context, details subapi.SubscriptionDet
 type subContext struct {
 	*e2Client
 	sub    *subapi.Subscription
+	errCh  chan error
 	cancel context.CancelFunc
 }
 
-func (s *subContext) ID() subapi.ID {
-	return s.sub.ID
+func (c *subContext) ID() subapi.ID {
+	return c.sub.ID
+}
+
+func (c *subContext) Err() <-chan error {
+	return c.errCh
 }
 
 // subscribe activates the subscription context
```

Comment:
@kuujo 
there is one minor problem that should be fixed. This condition should be changed to consider failures cases otherwise it skips the event

```
if event.Task.EndpointID == prevEndpoint {
continue
}
```

Perhaps a condition should be added to check if the event failure is nil and current end point and previous one are the same.